crates/teloxide/examples/chat_member_updates.rs

```diff
@@ -0,0 +1,92 @@
+// Import Teloxide and dependencies modules
+use teloxide::prelude::*;
+use teloxide::types::{ParseMode, ParseMode::MarkdownV2, ChatMemberUpdated, Update};
+use teloxide::dispatching::Dispatcher;
+use teloxide_core::adaptors::DefaultParseMode;
+
+// We create a type alias for our bot type
+type Bot = DefaultParseMode<teloxide::Bot>;
+
+// Main function that starts the bot
+#[tokio::main]
+async fn main() -> ResponseResult<()> {
+
+    // Initialize the logging environment for Teloxide
+    pretty_env_logger::init();
+
+    // Print a message to the console
+    log::info!("Starting Welcome/Goodbye bot...");
+
+    // in this case we use env variables to get the bot token
+    let bot = teloxide::Bot::from_env().parse_mode(MarkdownV2); // We use parse_mode method for use MarkdownV2 by default
+
+    // We create a handler for our botc
+    let handler = dptree::entry().inspect(|u:Update| {
+        println!("{u:#?}"); // Print the update to the console with inspect method and a closure
+    })
+        .branch(Update::filter_chat_member() // We use filter_chat_member method for filter the updates
+            .filter(|m: ChatMemberUpdated| // Don't forget to use the filter method for filter the updates
+                m.old_chat_member.kind.is_left() && // In this case, is a welcome message branch
+                    m.new_chat_member.kind.is_present())
+            .endpoint(chat_member_welcome))
+
+        .branch(Update::filter_chat_member() // We use filter_chat_member method for filter the updates
+            .filter(|m: ChatMemberUpdated| // Don't forget to use the filter method for filter the updates
+                m.new_chat_member.kind.is_left() && // In this case, is a goodbye message branch
+                    m.old_chat_member.kind.is_present())
+            .endpoint(chat_member_goodbye));
+
+    // We create a dispatcher for our bot
+    Dispatcher::builder(bot, handler)
+        .enable_ctrlc_handler()
+        .build()
+        .dispatch()
+        .await;
+
+    Ok(())
+}
+
+// Welcome Function
+// We use ChatMemberUpdated instead of Message for our function because
+// Chat member updates != messages
+// We use ParseMode::Html because to something users can use
+// "_" and "*" on their names and cause conflict with MarkdownV2
```

Comment:
This should probably be a doc comment?...

Reply:
Fixed